src/Servers/Kestrel/Core/src/Internal/HttpConnection.cs

```diff
@@ -130,9 +131,6 @@ private void StopProcessingNextRequest()
 
                 switch (_protocolSelectionState)
                 {
-                    case ProtocolSelectionState.Initializing:
```

Comment:
nit: maybe have a default case which does `Debug.Assert(false)`

Reply:
I'm definitely not a fan of not handling every case explicitly. Had I not been handling every case, I would have definitely added a default case that threw or something. I probably should have done that anyway.

Reply:
Can we get rid of the Initializing state now?

Reply:
It’s might be useful for a dump if we ever hit the abort case. Though let me ask, so we need to abort at all? Can we just unwind the loop?

Reply:
I'm not sure I understand the question. If we don't have a way to abort a request where the client is reading a response to slowly, how do we expect the request processing loop to unwind?

Reply:
Sorry I meant the case where an invalid protocol was selected. It seems like unwinding wouldn’t just fine. No need to abort

Reply:
In the case of protocol selection failing, since SelectProtocol() logs the issue anyway, I guess that is as long as we didn't try to write anything to the output. Otherwise, backpressure from the client could stop the socket from closing since no drain timeout would be configured on that path.

Reply:
Nothing can write, we’re the sole owner of the transport,  nothing is written so there’s nothing to drain

Reply:
I might leave a comment warning about the need to abort or set a drain timeout if anything was written to the output, but removing the abort seems fine in that case.

Reply:
I think we also need to get rid of IConnectionTimeoutFeature. The HttpsAdapter used to use it, but now I think it's broken for connection middleware since it's all being handled at the HttpConnection layer which was below the old connection adapters, but above the new connection middleware.

Reply:
Agreed. It may be interesting to move the drain timeout out of the http layer. It’s odd that we’re not undoing it (though you can’t undo anything because of the OnHeartbeat API) when we exit the middleware execution